modules/fedimint-wallet-server/src/lib.rs

```diff
@@ -1305,8 +1289,58 @@ struct StatelessWallet<'a> {
 }
 
 impl<'a> StatelessWallet<'a> {
+    /// Given a tx created from an `WalletOutput`, validate there will be no
+    /// issues submitting the transaction to the Bitcoin network
+    fn validate_tx(
+        &self,
+        tx: &UnsignedTransaction,
+        output: &WalletOutput,
```

Comment:
It feels a bit wrong making the `StatelessWallet` aware of of the wallet module, originally it was intended as a place to put all the BTC logic that can be tested separately. This is still kinda the case since `WalletOutput` is only a data struct and I don't have a better solution to offer rn :/

Reply:
> originally it was intended as a place to put all the BTC logic that can be tested separately

This is still true, but perhaps we need an easier way of unit testing the module functions.  `FakeFed` is a little too involved for unit testing.